libtelemetry/src/test/java/com/mapbox/services/android/telemetry/MapboxTelemetryTest.java

```diff
@@ -458,19 +498,219 @@ public void checksStopServiceWhenOptedOut() throws Exception {
+    TelemetryClient mockedTelemetryClient = mock(TelemetryClient.class);
+    Callback mockedHttpCallback = mock(Callback.class);
+    SchedulerFlusher mockedSchedulerFlusher = mock(SchedulerFlusher.class);
+    Clock mockedClock = mock(Clock.class);
+    LocalBroadcastManager mockedLocalBroadcastManager = mock(LocalBroadcastManager.class);
+    MapboxTelemetry theMapboxTelemetry = new MapboxTelemetry(mockedContext, aValidAccessToken, aValidUserAgent,
+      mockedEventsQueue, mockedTelemetryClient, mockedHttpCallback, mockedSchedulerFlusher, mockedClock,
+      mockedLocalBroadcastManager);
+    theMapboxTelemetry.enable();
+
+    String testUserAgent = "mapbox-navigation-ui-android/";
+    boolean userAgentBool = theMapboxTelemetry.checkRequiredParameters(aValidAccessToken, testUserAgent);
+
+    assertTrue(userAgentBool);
+  }
+
+  @Test
+  public void checksUserAgentEvents() throws Exception {
+    Context mockedContext = mock(Context.class);
+    String aValidAccessToken = "validAccessToken";
+    String aValidUserAgent = "MapboxEventsAndroid/";
+    EventsQueue mockedEventsQueue = mock(EventsQueue.class);
+    TelemetryClient mockedTelemetryClient = mock(TelemetryClient.class);
+    Callback mockedHttpCallback = mock(Callback.class);
+    SchedulerFlusher mockedSchedulerFlusher = mock(SchedulerFlusher.class);
+    Clock mockedClock = mock(Clock.class);
+    LocalBroadcastManager mockedLocalBroadcastManager = mock(LocalBroadcastManager.class);
+    MapboxTelemetry theMapboxTelemetry = new MapboxTelemetry(mockedContext, aValidAccessToken, aValidUserAgent,
+      mockedEventsQueue, mockedTelemetryClient, mockedHttpCallback, mockedSchedulerFlusher, mockedClock,
+      mockedLocalBroadcastManager);
+    theMapboxTelemetry.enable();
+
+    String testUserAgent = "MapboxEventsAndroid/";
+    boolean userAgentBool = theMapboxTelemetry.checkRequiredParameters(aValidAccessToken, testUserAgent);
+
+    assertTrue(userAgentBool);
+  }
+
+  @Test
+  public void checksInvalidUserAgent() throws Exception {
+    Context mockedContext = mock(Context.class);
+    String aValidAccessToken = "validAccessToken";
+    String aValidUserAgent = "MapboxEventsAndroid/";
+    EventsQueue mockedEventsQueue = mock(EventsQueue.class);
+    TelemetryClient mockedTelemetryClient = mock(TelemetryClient.class);
+    Callback mockedHttpCallback = mock(Callback.class);
+    SchedulerFlusher mockedSchedulerFlusher = mock(SchedulerFlusher.class);
+    Clock mockedClock = mock(Clock.class);
+    LocalBroadcastManager mockedLocalBroadcastManager = mock(LocalBroadcastManager.class);
+    MapboxTelemetry theMapboxTelemetry = new MapboxTelemetry(mockedContext, aValidAccessToken, aValidUserAgent,
+      mockedEventsQueue, mockedTelemetryClient, mockedHttpCallback, mockedSchedulerFlusher, mockedClock,
+      mockedLocalBroadcastManager);
+    theMapboxTelemetry.enable();
+
+    String aInvalidUserAgent = "invalidUserAgent";
```

Comment:
Same comments as `checksUserAgentTelemetryAndroid` test.

Reply:
Why not using `aValidUserAgent`? `aValidUserAgent` is redundant. We should fix the name if we want to remark something around the access token variable. `aInvalidUserAgent` is 👍